### DISCUSSION TITLE
flux bootstrap bitbucket-server status 401

### DISCUSSION THREAD

**Discussion by martin0598** [fluxcd/flux2#4005] [Q&A] [answered]
Hi there,

I'm just starting to try flux. Version is 2.0.0-rc.5. I'm running bitbucket server in a docker container, exposing https port to 8443. I hava a project 'Spielwiese' with the key 'SPIEL'.

According to https://fluxcd.io/flux/installation/#bitbucket-server-and-data-center I added an SSH key for my account in bitbucket. SSH access is enabled. The same key I added to env var BITBUCKET_TOKEN on my debian shell.

using following bootstrap command:

 flux bootstrap bitbucket-server \
    --owner=SPIEL  \
    --hostname=my-bb-server:8443 \
    --username=itsme \
    --repository=flux-fleet \
    --path=./clusters/mycluster/ \
    --branch=main \
    --personal 

I get:

► connecting to my-bb-server-02:8443
✗ failed to get Git repository "https:/my-bb-server:8443/SPIEL/flux-fleet": failed to get repository ~SPIEL/flux-fleet: get respository failed: request GET https://my-bb-server:8443/rest/api/1.0/projects/~SPIEL/repos/flux-fleet returned status code: 401 , unexpected status code

Any hints?
Best regards,
Martin

**Comment by kingdonb** [comment]
A 401 error means unauthorized, so the most likely cause is an issue with the credentials you provided.

You may have a bad secret, bootstrapping will only create a new secret if the `flux-system` secret is missing from the `flux-system` namespace. If you created it once, with the wrong BITBUCKET_TOKEN and it failed, you will need to delete that secret to try again.

If you try with normal SSH as a user (not through Flux) does it work? The access issue could come from a bad token, or from wrong permissions. The bootstrap user needs write access to the repo, and I'm not sure about bitbucket, but I think it also needs an admin access to the repo in order to set the access token. Let us know if any of this helps, or join the #Flux channel on the CNCF Slack for more real-time and interactive debugging help!

**Reply by martin0598** [reply, marked as answer]
Hi kingdonb,
thanks for your answer. I copied the wrong http access token to my environment. For other reasons I now take the generic git way, that seems to work fine for me (at first sight)
Best regards,
Martin
